packages/rocketchat-api/server/v1/chat.js

```diff
@@ -255,3 +255,23 @@ RocketChat.API.v1.addRoute('chat.update', { authRequired: true }, {
 		});
 	}
 });
+
+RocketChat.API.v1.addRoute('chat.react', { authRequired: true }, {
+	post() {
+		if (!this.bodyParams.messageId || !this.bodyParams.messageId.trim()) {
+			throw new Meteor.Error('error-messageid-param-not-provided', 'The required "messageId" param is missing.');
+		}
+
+		const msg = RocketChat.models.Messages.findOneById(this.bodyParams.messageId);
+
+		if (!msg) {
+			throw new Meteor.Error('error-message-not-found', 'The provided "messageId" does not match any existing message.');
+		}
+
+		const emoji = this.bodyParams.emoji;
+
+		Meteor.runAsUser(this.userId, () => Meteor.call('setReaction', emoji, msg._id, this.userId));
```

Comment:
Why do you need to pass the `this.userId` if you are already executing the method using runAsUser?

Reply:
@rodrigok you are right it is not strictly necessary to do it that way.
However if you look at every single method on that file, they all do it in that very same way.
So I just tried to stick to the standards and to the scope.

Reply:
@jgtoriginal What method(s) and what files are you talking about?

Reply:
@graywolf336 every API method on this file `packages/rocketchat-api/server/v1/chat.js` does `Meteor.runAsUser`. Does it make sense?

Reply:
Yes, that part makes sense however the part we are talking about is the passed in `this.userId` to the `Meteor.call('setReaction...` piece of the code along with the new argument `usr` added to the definition of the `setReaction` method. These last two aren't necessary and can be removed.